### PR TITLE
o/svcstate/svcstatetest, daemon/api_quotas: fix some tests, add mock helper

### PR DESCRIPTION
**Pull request by anonymouse64**
* Fix some of the tests in api_quotas_test which were not restoring mocked functions properly.
* Add servicestatetest.MockQuotaInState and use it in external package tests
* Also delete a completed TODO

### CODE REVIEW

File: overlord/servicestate/servicestatetest/quotas.go [hunk 1]
```diff
@@ -30,3 +34,24 @@ import (
 func PatchQuotas(st *state.State, grps ...*quota.Group) (map[string]*quota.Group, error) {
 	return internal.PatchQuotas(st, grps...)
 }
+
+func MockQuotaInState(st *state.State, quotaName string, parentName string, snaps []string, memoryLimit quantity.Size) error {
+	osutil.MustBeTestBinary("cannot mock quota group in state outside of tests")
```
Comment (pedronis) [inline overlord/servicestate/servicestatetest/quotas.go:39]: as this is a *test package we don't strictly need this

**Review by pedronis — COMMENTED** [review]
thanks

**Review by pedronis — APPROVED** [review]
thanks

File: overlord/servicestate/servicestatetest/quotas.go (hunk 1 above)
Comment (anonymouse64) [reply overlord/servicestate/servicestatetest/quotas.go:39]: since this PR is a skip spread one, I will push this change now

Comment (anonymouse64) [reply overlord/servicestate/servicestatetest/quotas.go]: done

**Review by mvo5 — APPROVED** [review]
Thank you

**Pull request merged**